ls1intum/Artemis
# `General`: Fix several small issues in different charts

### Checklist
#### General


#### Server
#### Client

### Motivation and Context


This PR contains several small improvements:

- **Course detail charts**: We faced some overlapping on production for the doughnut charts on course detail page due to large numbers displayed within the doughnut. These are fixed now.
- **Average student score pie chart**: The denominator representing the maximum of reachable points in the course was not correct. This is fixed now.
- **Exercise assessment dashboard pie chart**: If a programming exercise is corrected automatically, the chart currently represents all submissions as unassessed. For this case, the chart is adapted
- **Chart coloring**: Along Artemis, we have several occurrences of the same color, especially in charts, where the color codes are dedicated. I started to centralize the color definition.

### Description

- **Course detail charts**: All charts are now 50px bigger
- **Average student score pie chart**: I removed a occurrence of `.collect(Collectors.toSet())` that filtered all duplicated max points of exercises and therefore returned the wrong reachable points for a course.
- **Exercise assessment dashboard pie chart**: The chart is now representing the ratio of open and resolved complaints in this case as this is the only measurement of interest then.
- **Chart coloring**: I replaced some dedicated color codes with the enum `GraphColors`.
### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 Students
- 1 Programming Exercise with automatic assessment Complaints enabled

1. Log in to Artemis
2. Participate with the students on the Programming exercise
3. Assess the participation as instructor
4. Complain as one student
5. Navigate to the exercise assessment dashboard of the exercise and inspect the pie chart. It should only represent complaints.
6. Also check any other assessment dashboard and the course assessment dashboard. The legend entries should be as usual.
7. Check the course detail page and check the charts there. Make sure no overlapping occurs.

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage






### Screenshots


The mentioned overlapping:
![Bildschirmfoto 2022-02-07 um 22 58 29](https://user-images.githubusercontent.com/80622272/153206030-35a6d102-effd-4cb6-806d-51249843e8b5.png)

How the Pie charts look now:
![image](https://user-images.githubusercontent.com/80622272/153258908-667742a9-779a-4935-ac65-bc64269e3a6b.png)

The old denominator (that is incorrect):
![image](https://user-images.githubusercontent.com/80622272/153259463-f7c555a9-ace3-45b3-ac5f-8c20d5e93647.png)

The new Pie chart colors (these are open for discussion):
![image](https://user-images.githubusercontent.com/80622272/153258426-2680f8af-5127-4bbf-9811-3ccaf86c0e70.png)

How the Pie chart looks in case of automatic assessment programming exercise with complaints enabled:
![image](https://user-images.githubusercontent.com/80622272/153258707-4cb9eadc-1308-4e97-8c5d-d666b9755f49.png)
